src/plugins/template/tests/functional/op_reference/sign.cpp

```diff
@@ -66,6 +66,11 @@ INSTANTIATE_TEST_SUITE_P(smoke_Sign_With_Hardcoded_Refs,
                                                       element::f32,
                                                       std::vector<float>{1, -2, 0, -4.8f, 4.8f, -0.0f},
                                                       std::vector<float>{1, -1, 0, -1, 1, 0}),
+                                           SignParams(PartialShape{7},
+                                                      element::f32,
+                                                      element::f32,
+                                                      std::vector<float>{1, -2, 0, NAN, -4.8f, 4.8f, -0.0f},
```

Comment:
Use `std::numeric_limits<float>` for NAN values instead using macro `NAN`. This macro is less reliable than `std` in some case on some platforms can gives different values.

Apply  change in all modified/created test cases

Reply:
std::numeric_limits function is used in modified/created test cases of this PR